packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -421,9 +400,43 @@ export class AwsCustomResource extends Construct implements iam.IGrantable {
       },
     });
 
-    // If the policy was deleted first, then the function might lose permissions to delete the custom resource
-    // This is here so that the policy doesn't get removed before onDelete is called
-    this.customResource.node.addDependency(policy);
+    // Create the policy statements for the custom resource function role, or use the user-provided ones
```

Comment:
Why move the location of this?

Reply:
The old code created the policy first, followed by the custom resource, and finally made the custom resource dependent on the policy. It felt a bit more natural to first create the custom resource, and then create the policy + dependency on the policy together in the same `if(props.policy) { ... }` block.

The alternative is to not move this around and wrap both the policy creation and dependency addition with `if(props.policy)`, but it feels neater in one block.